uberfire-structure/uberfire-structure-api/src/main/java/org/guvnor/structure/pom/DependencyType.java

```diff
@@ -17,6 +17,7 @@
 /***
  * Types managed by the Pom's event
  */
-public enum DependencyType {
-    JPA
+public interface DependencyType {
```

Comment:
@desmax74 @porcelli @ederign  if for extensibility reasons we're switching to this inheritance-based solution for dependency types (instead of ENUM based solution), does it make sense to keep the "data" associated with each dependency type in one central `DependencyTypesMapper.json` file?
Wouldn't it be better to just have `DependencyType` have more methods, like `Set<GAV> getDependencies()` etc?
In that case we could get rid of all the code loading those data from json and mapping etc and the information for each dependency type would be more "centralized". What do you think?

Reply:
@jhrcek you're absolutely right! the only challenge is the V (version) of the GAV, that should be collected from our existing POM variables

Reply:
@jhrcek @porcelli the informations loaded from the json and from the Pom for each dependency are already contained in the DynamicDependencyPom https://github.com/kiegroup/appformer/pull/489/files#diff-22d70676e71cb2474a5c70629db02303 
the DependencyType works just as key in the ConfigurationMap https://github.com/kiegroup/appformer/pull/541/files#diff-eb740a937fd6b8ef9c34163fbc462c84

Reply:
@desmax74 could you point the json config files? as @jhrcek mentioned, we likely could have the GA hard coded in Java classes directly, and just collecting the V (version) from a config file that ideally should be fed by POM vars

Reply:
@porcelli here the file with the placeholders https://github.com/kiegroup/appformer/blob/1f6b657d1ec89344bc14dcb95be4720505c3beb2/uberfire-structure/uberfire-structure-backend/src/main/filtered-resources/DependencyTypesMapper.json

Reply:
The files contains the common (except SCENARIO_RUNTIME) dependencies like internals, test and validation, every module own a dedicated file when will be ready the infrastructure. Using a placeholder inside class means write a class for every dependency needed by each editor, instead to write one line in the config, if I have understand correctly your proposal.

Reply:
every editor should provide an interface that would provide a list of GAVs dependencies, and in this file that a specific editor would need to provide you could have multiple placeholder inside.. but still one interface `per editor` or per component.

Reply:
@porcelli ok every impl type (like JPA https://github.com/kiegroup/appformer/blob/1f6b657d1ec89344bc14dcb95be4720505c3beb2/uberfire-structure/uberfire-structure-api/src/main/java/org/guvnor/structure/pom/types/JPADependencyType.java) will contains the dependencies informations

Reply:
@porcelli sound good now with the placeholder in the java templates replaced before the build ?

Reply:
It's fine but in this way the Config become a registry (is already present an addDependencies) for the others modules' dependencies, is this ok ?

Reply:
Done

Reply:
@porcelli is it ok with the latest changes ?